cost-analyzer/templates/_helpers.tpl

```diff
@@ -1006,6 +1006,44 @@ Begin Kubecost 2.0 templates
       value: "false" # this container should never run KC's concept of "ETL"
     - name: CLOUD_PROVIDER_API_KEY
       value: "AIzaSyDXQPG_MHUEy9neR7stolq6l0ujXmjJlvk" # The GCP Pricing API key.This GCP api key is expected to be here and is limited to accessing google's billing API.'
+    {{- if .Values.global.integrations.postgres.enabled }}
+    - name: AGGREGATOR_ADDRESS
+    {{- if or .Values.saml.enabled .Values.oidc.enabled }}
+      value: localhost:9008
+    {{- else }}
+      value: localhost:9004
+    {{- end }}
+    - name: INT_PG_ENABLED
+      value: "true"
+    - name: INT_PG_RUN_INTERVAL
+      value: {{ quote .Values.global.integrations.postgres.runInterval }}
+    - name: INT_PG_DB_HOST
+      value: {{ quote .Values.global.integrations.postgres.databaseHost }}
+    - name: INT_PG_DB_PORT
+      value: {{ quote .Values.global.integrations.postgres.databasePort }}
+    - name: INT_PG_DB_NAME
+      value: {{ quote .Values.global.integrations.postgres.databaseName }}
+    - name: INT_PG_DB_USER
+      value: {{ quote .Values.global.integrations.postgres.databaseUser }}
+    - name: INT_PG_DB_PASS
+      value: {{ quote .Values.global.integrations.postgres.databasePassword }}
```

Comment:
I know you're aware of this, but it still makes me uncomfortable that a plaintext password would end up in someone's Helm values. Should we be shipping a feature that encourages a bad security practice like that? And more practically, will anyone adopt the feature in this state?

Reply:
Definitely agree that this isn't the right long term approach. I had prototyped a minor improvement by doing:

```yaml
    - name: INT_PG_DB_PASS
      valueFrom:
        secretKeyRef:
          name: pg-pass
          key: password
```

however, even this wouldn't be the fully correct solution (for reasons Chip mentioned above). I'm a bit wary to deliver this half baked solution, then have to walk it back and replace it with the fully correct solution. I think that we can get a fully correct solution to build nicely on top of these configs and be made available in a patch release or v2.3? Thoughts?

@chipzoller Let me know what you think too! If we're all ok with this, I'd love to merge and test. In the meantime, I'll be working on the full solution here.

Reply:
Unless there is, for some reason, an absolute hard requirement this _must_ be in 2.2, I would prefer we hold until 2.3 where we get the security aspect a little better so we avoid breaking changes right out of the gate. And as we have already passed the agreed upon go/no-go time, I think it's the better thing to do. Releasing this in a patch to 2.2 would not really be appropriate as this is a definite feature.